packages/common-components/src/TextInput/FormikTextInput.tsx

```diff
@@ -21,14 +22,15 @@ const FormikTextInput = React.forwardRef(
       disabled = false,
       autoFocus,
       captionMessage,
+      hideLabel,
       ...rest
     }: FormikTextInputProps,
     forwardedRef: any
   ) => {
     const [field, meta, helpers] = useField(name)
     return (
       <TextInput
-        label={label ? label : field.name}
+        label={label ? label : hideLabel ? undefined : field.name}
```

Comment:
This is convoluted. Would this make more sense?
```suggestion
        label={hideLabel ? undefined : label || field.name}
```